packages/react-scripts/config/webpack.config.prod.js

```diff
@@ -250,6 +265,41 @@ module.exports = {
     ],
   },
   plugins: [
+    // configuration for vendor splitting and long term caching
+    // more info: https://medium.com/webpack/predictable-long-term-caching-with-webpack-d3eee1d3fa31
+    new webpack.NamedModulesPlugin(),
```

Comment:
just in case - webpack docs recommend adding also `HashedModuleIdsPlugin` for prod build. https://webpack.js.org/guides/caching/#deterministic-hashes

Reply:
That plugin appears to be an alternative to `NamedModulesPlugin` and `NamedChunksPlugin`. While `HashedModuleIdsPlugin` requires less configuration, it can result in significantly larger chunks after gzip (webpack/webpack.js.org#1597 mentions ~9% reduction in chunk size when switching from Hashed to Named, and I've observed similar size discrepancies as well).